Patch-Source: https://github.com/Cacti/cacti/commit/71e734c439c5e4e1fc3e484d704524b31fa99030
From 71e734c439c5e4e1fc3e484d704524b31fa99030 Mon Sep 17 00:00:00 2001
From: TheWitness <thewitness@cacti.net>
Date: Fri, 1 Jul 2022 16:47:48 -0400
Subject: [PATCH] Fixing #4853 - Realtime Warnings PHP8.1

Deprecation warnings when attempting real-time Graphs with PHP8.1
---
 CHANGELOG   | 1 +
 lib/rrd.php | 2 +-
 2 files changed, 2 insertions(+), 1 deletion(-)

diff --git a/lib/rrd.php b/lib/rrd.php
index 04216deba..d6897c400 100644
--- a/lib/rrd.php
+++ b/lib/rrd.php
@@ -3669,7 +3669,7 @@ function rrdtool_create_error_image($string, $width = '', $height = '') {
 
 	/* see the size of the string */
 	$string    = trim($string);
-	$maxstring = (450 - (125 + 10)) / ($font_size / 0.9);
+	$maxstring = ceil((450 - (125 + 10)) / ($font_size / 0.9));
 	$stringlen = strlen($string) * $font_size;
 	$padding   = 5;
 
